integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -179,7 +187,15 @@ def from_dict(cls, data: Dict[str, Any]) -> "GoogleAIGeminiChatGenerator":
         deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
 
         if (tools := data["init_parameters"].get("tools")) is not None:
-            data["init_parameters"]["tools"] = [Tool.deserialize(t) for t in tools]
+            deserialized_tools = []
```

Comment:
Do we want to add a similar check here to prevent crashing in case of errors?
if isinstance(tool, Tool):

Reply:
It's not necessary in here as we always convert `Tool`s to `ToolProto` before serialization.

Also what we receive here is just `bytes`.